backends/dnf/pk-backend-dnf.c

```diff
@@ -823,6 +823,25 @@ dnf_utils_run_query_with_newest_filter (DnfSack *sack, HyQuery query)
 	return results;
 }
 
+static gboolean
+dnf_utils_force_distupgrade_on_upgrade (DnfSack *sack)
+{
+	HyQuery query = hy_query_create(sack);
+	query->installed();
+	const gchar* distroverpkg_names[] = { "system-release", "distribution-release", NULL };
+	hy_query_filter_provides_in(query, (gchar**) distroverpkg_names);
+	const gchar* distupgrade_provides[] = { "system-upgrade(dsync)", "product-upgrade() = dup", NULL };
+	hy_query_filter_provides_in(query, (gchar**) distupgrade_provides);
+	GPtrArray *plist = hy_query_run(query);
+	gint candidates = plist->len;
+	g_ptr_array_unref(plist);
+
+	if (candidates > 0) {
+		return TRUE;
+	}
+	return FALSE;
```

Comment:
```suggestion
	return candidates > 0;
```

no need to do `if (X) return TRUE; return FALSE;`

Reply:
Do we do this normally anywhere else in the codebase? I didn't see this style of implicit conditional return being used elsewhere.

Reply:
Other parts of this file don't do that, so I'm keeping this as is...